.github/workflows/build-runner.yml

```
@@ -0,0 +1,47 @@
+name: Build runner
+
+on:
+  workflow_dispatch:
+    inputs:
+      bundle-tag:
+        description: 'Tag of the bundle release (e.g., "codeql-bundle-20200826")'
+        required: true
+
+jobs:
+  build-runner:
+    runs-on: ubuntu-latest
+    env:
+      RELEASE_TAG: "${{ github.event.inputs.bundle-tag }}"
+
+    strategy:
+      matrix:
+        extension: ["linux", "macos", "win.exe"]
+
+    steps:
+      - uses: actions/checkout@v2
```

```suggestion
      - uses: actions/checkout@v2
        with:
          ref: refs/tags/${{ github.event.inputs.bundle-tag }}
```

It's potentially confusing to build the code from `main` and upload it as an artifact on a tagged release.

(It would be nice if we could just use the workflow from the release tag too, but workflows can't be dispatched on tags currently.)

---

I've followed Robert's suggestion to make the bundle-tag optional, which now makes this a bit awkward. Do you see a way apart from having two calls to `actions/checkout@2` each with an `if`?

---

In the interests of getting a version of this ready I am going to merge. We can later update this if we see a way to resolve the inconsistency.